### ISSUE TITLE
Saving HTML outout

### ISSUE THREAD

**Issue by mglaman** [mglaman/drupal-test-helpers#4]

**Comment by mglaman** [comment]
@joachim-n I looked into this. There is _a lot_ of code to copy. I wonder if maybe it's best to document this and prove it in a test. If Drupal core moved the main callback of `\Drupal\Tests\BrowserHtmlDebugTrait::getResponseLogHandler` to a callable, then we could convert the Symfony response to a PSR-7 response and let the callable handle it.

**Comment by joachim-n** [comment]
I wrote some code for this... I can't remember where it is now unfortunately :(

It felt like I was copying a LOT of code from the same functionality in Drupal Functional tests!

**Comment by joachim-n** [comment]
Ah here it is: https://github.com/mglaman/drupal-test-helpers/commit/2dbed403a1872a3aea4b2d852a853d2616a84f16

**Comment by mglaman** [comment]
Yeah, it is a lot to figure out. Hm. Maybe this library could provide a trait that wraps `BrowserHtmlDebugTrait` to make it easier to leverage the Guzzle handler without actually using Guzzle.

Then we're not copying all of that code but making it easier to automatically save HTML output on `doRequest` or whenever asked to.

**Comment by joachim-n** [comment]
> Yeah, it is a lot to figure out. Hm. Maybe this library could provide a trait that wraps BrowserHtmlDebugTrait to make it easier to leverage the Guzzle handler without actually using Guzzle.

You mean do something like switch the part of Guzzle that does the actual HTTP request? That would be nifty. But maybe end up being expensive in terms of performance if we end up doing too much in and around Guzzle. The nice thing about calling the HTTP kernel directly is there are fewer layers involved. 

I was thinking about maybe doing a patch to core to refactor out a trait for writing HTML output.
